src/hyp3_sdk/jobs.py

```diff
@@ -127,6 +127,7 @@ def download_files(self, location: Path | str = '.', create: bool = True) -> lis
         if not self.succeeded():
             raise HyP3SDKError(f'Only succeeded jobs can be downloaded; job is {self.status_code}.')
         if self.expired():
+            assert isinstance(self.expiration_time, datetime)
```

Comment:
I think I slightly prefer `assert x is not None` for cases like this because it makes it more clear that that's what we're guarding against (rather than the variable being some other type), but I wouldn't block the PR for it.